UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -439,10 +440,13 @@ protected async Task ConstructImageBuffer(GLTFTexture texture, int textureIndex)
 				}
 			}
 
-			_assetCache.TextureCache[textureIndex] = new TextureCacheData
+			if (_assetCache.TextureCache[textureIndex] == null)
 			{
-				TextureDefinition = texture
-			};
+				_assetCache.TextureCache[textureIndex] = new TextureCacheData
+				{
+					TextureDefinition = texture
+				};
+			}
 		}
```

Comment:
Should this throw in the else case?  Or warn or something?  Or is it ok to silently do nothing on multiple calls to this function with the same textureIndex?

Reply:
Really this entire method needs work.  From what it looks like, this will fill in either ImageStreamCache or BufferCache depending on what type it is, but the IF statements dont make that intention clear.  Both of these are guarded against multiple calls with the same ID.  This also new's TextureCacheData, but that wasnt guarded (which was causing duplicate textures to be created).  I'm not entirely sure why this function gets called multiple times, but subsequent calls with the same ID do in fact intend to populate with the same values.

As for your comment, this frequently gets called multiple times with the same ID, so a warning would probably not be the right thing to do in this case.  If we rewrote this so that it only got called once per texture, I would be all for it.  (similar to the debug asserts I added elsewhere)